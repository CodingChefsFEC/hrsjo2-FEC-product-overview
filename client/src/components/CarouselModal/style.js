import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';
const superLightGray = 'rgb(244, 244, 244)'

export const Container = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
  background-color: white;

  .topRow {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 0.5rem 4rem;
    height: 88%;
  }

  .fa-times-circle {
    position: fixed;
    top: 3rem;
    right: 4rem;
  }
  .fa-times-circle:hover {
    cursor: pointer;
  }
`;

export const Thumbnails = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  margin-bottom: 0.5rem;
`;

export const Thumbnail = styled.div`
    margin: 0 5px;
    border: ${props => props.selected ? '2px' : '1px'} solid ${lightGray};
    overflow: hidden;
    height: 7.7vh;
    width: 5%;
    opacity: ${props => props.selected ? '1' : '0.25'};
    cursor: ${props => props.selected ? 'default' : 'pointer'};

    background-image: url("${props => props.image}");
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;
    background-color: ${props => props.image ? 'transparent' : superLightGray};

    display: flex;
    justify-content: center;
    align-items: center;

    .share-pic-text {
      font-size: 8px;
      font-weight: bold;
      color: black;
    }

`;

export const MainImage = styled.div`
  height: 100%;
  min-width: 40%;
  max-width: 80%;

  background-image: url("${props => props.image}");
  background-color: ${props => props.image ? 'transparent' : superLightGray};
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
`;

export const PhotoUpload = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;

  .photo-upload--text {
    margin: 0 10rem;
  }
  .photo-upload--button {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 1rem;
    background-color: white;
    padding: 0.6rem;
    border: 1px solid ${lightGray};
    border-radius: 3px;
    cursor: pointer;
  }
  .photo-upload--button:hover {
    background-color: ${superLightGray};
  }
  .camera-icon {
    margin-right: .6rem;
  }
`;