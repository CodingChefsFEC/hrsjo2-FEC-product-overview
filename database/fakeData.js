const fakeData = JSON.parse(`{
  "_id": {
    "$oid": "5fae237faafe544c4495f638"
  },
  "breadcrumbs": [
    "Target",
    "Bikes",
    "Chipss"
  ],
  "images": [
    "http://placeimg.com/640/480/food",
    "http://placeimg.com/640/480/abstract",
    "http://placeimg.com/640/480/fashion"
  ],
  "id": 0,
  "name": "Gorgeous Frozen Car",
  "brand": "Bergnaum Group",
  "price_reg": 50.37,
  "price_discount": 415.43,
  "sale_end": {
    "$date": "2021-08-27T12:13:01.224Z"
  },
  "total_reviews": 112,
  "total_questions": 6,
  "ratings_data": {
    "_id": {
      "$oid": "5fae237faafe544c4495f639"
    },
    "five_star": 96,
    "four_star": 9,
    "three_star": 12,
    "two_star": 33,
    "one_star": 39
  },
  "promo_data": {
    "_id": {
      "$oid": "5fae237faafe544c4495f63a"
    },
    "id": 4,
    "icon_img": "http://placeimg.com/640/480",
    "main_text": "Sit quo aut distinctio veniam.",
    "sub_text": "Sed nulla nihil modi non quia eligendi quia dignissimos.",
    "link_text": "Learn More"
  },
  "__v": 0
}`);

export default fakeData;