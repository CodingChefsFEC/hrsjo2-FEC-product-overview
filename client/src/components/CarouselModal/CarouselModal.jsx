import React, {useState} from 'react';
import { Container, Thumbnails, Thumbnail, MainImage } from './style.js';

const CarouselModal = ({ images, toggleCarousel }) => {
  // const [mainImage, setMainImage] = useState(images[0]);
  const [imageIndex, setImageIndex] = useState(0)

  // const changeImage = (e) => {
  //   const newImage = e.target.src;
  //   setMainImage(newImage);
  // }

  const changeImageIndex = (i) => {
    setImageIndex(i);
  }

  const nextImageIndex = () => {
    if (imageIndex < images.length - 1) {
      setImageIndex(imageIndex + 1);
    }
  }

  const prevImageIndex = () => {
    if (imageIndex > 0) {
      setImageIndex(imageIndex - 1);
    }
  }

  return (
    <Container>
      <div className="topRow">
        <span className="fa-stack fa-2x" onClick={prevImageIndex}>
          <i className="far fa-circle fa-stack-2x"></i>
          <i className="fas fa-chevron-left fa-stack-1x"></i>
        </span>
        <MainImage src={images[imageIndex]} />
        <span className="fa-stack fa-2x" onClick={nextImageIndex}>
          <i className="far fa-circle fa-stack-2x"></i>
          <i className="fas fa-chevron-right fa-stack-1x"></i>
        </span>
      </div>
      <Thumbnails>
        {images.map((image, i) => {
          return (
            <Thumbnail src={image}
              onClick={() => {
                changeImageIndex(i);
              }}
            />
          );
        })}
      </Thumbnails>
      <i className="fas fa-times-circle fa-4x" onClick={toggleCarousel}></i>

    </Container>
  );
};

export default CarouselModal;