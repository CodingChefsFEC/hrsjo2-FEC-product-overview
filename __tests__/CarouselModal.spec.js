import React from 'react';
import { create } from 'react-test-renderer';
import CarouselModal from '../client/src/components/CarouselModal/CarouselModal.jsx';
import fakeData from '../database/fakeData.js';

// const getProductData = () => {
//   axios.get(`/api/products${window.location.pathname}`)
//     .then(({data}) => {
//       console.log(data);
//       setProductData(data);
//     });
// }

// const toggleCarousel = () => {
//   setCarousel(!carousel);
// }

// Snapshot test
describe("CarouselModal component", () => {
  test("it displays the next image", () => {
    const component = create(<CarouselModal
      images={fakeData.images}
    />);
    expect(component.toJSON()).toMatchSnapshot();
  })
});

// Snapshot test
// describe("CarouselModal component", () => {
//   test("it displays the next image", () => {
//     const component = create(<CarouselModal
//       images={fakeData.images}
//       toggleCarousel={toggleCarousel}
//     />);
//     const instance = carouselModal.root;
//     const carouselModal = instance.findByType("img");

//     expect(component.toJSON()).toMatchSnapshot();
//   })
// });