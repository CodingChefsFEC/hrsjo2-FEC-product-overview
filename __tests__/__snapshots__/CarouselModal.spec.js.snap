// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselModal component it displays the next image 1`] = `
<div
  className="style__Container-sc-1jgu9eh-0 hirjaI"
>
  <div
    className="topRow"
  >
    <span
      className="fa-stack fa-2x"
      onClick={[Function]}
    >
      <i
        className="far fa-circle fa-stack-2x"
      />
      <i
        className="fas fa-chevron-left fa-stack-1x"
      />
    </span>
    <img
      className="style__MainImage-sc-1jgu9eh-3 iBEeis"
      src="http://placeimg.com/640/480/food"
    />
    <span
      className="fa-stack fa-2x"
      onClick={[Function]}
    >
      <i
        className="far fa-circle fa-stack-2x"
      />
      <i
        className="fas fa-chevron-right fa-stack-1x"
      />
    </span>
  </div>
  <div
    className="style__Thumbnails-sc-1jgu9eh-1 lfkSAx"
  >
    <img
      className="style__Thumbnail-sc-1jgu9eh-2 eKEUVQ"
      onClick={[Function]}
      src="http://placeimg.com/640/480/food"
    />
    <img
      className="style__Thumbnail-sc-1jgu9eh-2 eKEUVQ"
      onClick={[Function]}
      src="http://placeimg.com/640/480/abstract"
    />
    <img
      className="style__Thumbnail-sc-1jgu9eh-2 eKEUVQ"
      onClick={[Function]}
      src="http://placeimg.com/640/480/fashion"
    />
  </div>
  <i
    className="fas fa-times-circle fa-4x"
  />
</div>
`;
